doc/blog/201809-UnoUnderTheHoodXamlGeneration.md

```diff
@@ -0,0 +1,84 @@
+
+All this happens behind the scenes whenever you build an Uno head. 
+
+## The end result
+
+Let's look at a concrete example. This is part of the markup we put in `MainPage.xaml` in a previous article: 
+
+```` xml
+    <StackPanel> 
+        <TextBlock x:Name="ClickTextBlock" 
+                   Text="Button wasn't clicked yet" /> 
+        <Button Content="Click me" 
+                Click="Button_Click" /> 
+    </StackPanel> 
+````
+
+The generated output can be found in the 'obj' folder under the project head. 
+
+![Xaml Code Generation Location](Assets/xaml-code-generation-location.png)
+*The code file generated from MainPage.xaml.*
+
+This is a part of the generated code, showing the output for the XAML snippet above: 
+
+```` csharp
+    public sealed partial class MainPage : Windows.UI.Xaml.Controls.Page 
+    { 
+        private void InitializeComponent() 
+        { 
+            var nameScope = new global::Windows.UI.Xaml.NameScope(); 
+            NameScope.SetNameScope(this, nameScope); 
+            // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 1:2) 
+            Content =             new global::Windows.UI.Xaml.Controls.StackPanel 
+            { 
+                // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 8:3) 
+                Children =  
+                { 
+                    new global::Windows.UI.Xaml.Controls.TextBlock 
+                    { 
+                        Name = "ClickTextBlock", 
+                        Text = "Button wasn't clicked yet", 
+                        // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 9:4) 
+                    } 
+                    .MainPage_05a03e67d6a6581089c5d12899278d61_XamlApply((MainPage_05a03e67d6a6581089c5d12899278d61XamlApplyExtensions.XamlApplyHandler0)(c0 =>  
+                    { 
+                        nameScope.RegisterName("ClickTextBlock", c0); 
+                        this.ClickTextBlock = c0; 
+                    } 
+                    )) 
+                    , 
+                    new global::Windows.UI.Xaml.Controls.Button 
+                    { 
+                        Content = @"Click me", 
+                        // Source ..\..\..\..\..\..\UnoExtTestbed.Shared\MainPage.xaml (Line 11:4) 
+                    } 
+                    .MainPage_05a03e67d6a6581089c5d12899278d61_XamlApply((MainPage_05a03e67d6a6581089c5d12899278d61XamlApplyExtensions.XamlApplyHandler1)(c1 =>  
+                    { 
+                        c1.Click += Button_Click; 
+                    } 
+                    )) 
+                    , 
+                } 
+            }
+````
+
+What a mouthful! In the normal course of Uno development you don't need to look at code generated from XAML or even know that it's there, though occasionally it's useful to take a peek when debugging.
```

Comment:
One key aspect of this code is that it's meant to be machine optimized, not human reader optimized. For instance, the cast to XamlApplyHandler may appear unnecessary, but it's actually based on optimization to help the Roslyn compiler to avoid having to do very expensive computations to try and determine that very type we can provide.  It basically slashed in half the compilation time of large Xaml files.

Reply:
Nice detail, I'll add it :)